ni_measurementlink_service/_internal/parameter/serialization_strategy.py

```diff
@@ -1,25 +1,24 @@
 """Serialization Strategy."""
 
-from typing import Any, Callable
+from typing import Any, Callable, Dict
 
 from google.protobuf import type_pb2
-from google.protobuf.internal import decoder
-from google.protobuf.internal import encoder
+from google.protobuf.internal import decoder, encoder
+from typing_extensions import TypeAlias
 
+InnerEncoder: TypeAlias = Callable[[Callable[[bytes], int], bytes, bool], int]
+InnerDecoder: TypeAlias = Callable[
+    [memoryview, int, int, type_pb2.Field.Kind.ValueType, Dict[int, Any]], int
+]
+EncoderFactory: TypeAlias = Callable[[int], InnerEncoder]
+DecoderFactory: TypeAlias = Callable[[int, str], InnerDecoder]
 
-def _scalar_encoder(encoder) -> Callable[[int], Callable]:
-    """Abstract Specific Encoder(Callable) as Scalar Encoder Callable that takes in field index.
 
-    Args
-    ----
-       encoder (Callable[[int, bool, bool], Callable]): Specific encoder that takes in
-       field_index, is_repeated, is_packed and returns the Low-level Encode Callable.
-
-    Returns
-    -------
-        Callable[[int],Callable]: Callable Encoder for scalar types that takes
-        in field_index and returns the Low-level Encode Callable.
+def _scalar_encoder(encoder) -> Callable[[int], InnerEncoder]:
+    """Constructs a scalar encoder factory that takes a field index and returns an InnerEncoder.
 
+    This class returns the InnerEncoder callable with is_repeated set to False
+    and is_packed set to False.
     """
 
     def scalar_encoder(field_index):
```

Comment:
Untyped return values are also treated as `Any`. When you add type hints to the outer function, the inner function will still allow type mismatches.

Reply:
Added type hints for this inner method.